stock_helper/models/stock_location.py

```diff
@@ -18,3 +19,28 @@ def is_sublocation_of(self, others, func=any):
         # Efficient way to verify that the current location is
         # below one of the other location without using SQL.
         return func(self.parent_path.startswith(other.parent_path) for other in others)
+
+    @api.returns("stock.warehouse", lambda value: value.id)
+    def get_closest_warehouse(self):
+        """Returns warehouse id of the closest warehouse that contains location
+
+        By default the get_warehouse (which is located in the odoo core module stock)
+        returns the warehouse via searching the view_location_id with parent_of.
+        If we have multiple warehouses
+        where a view_location_id is a parent of an other view_location_id
+        Then it depends on the sorting of the warehouses which this method returns
+
+        With this methods we will realy get the closest warehouse of a location
+        """
+        warehouses = self.env["stock.warehouse"].search([])
+        res = warehouses.browse()
+        for location in self:
```

Comment:
on how many records do you expect to call this? Judging from the docstring "contains location" I expect you call it on just one location.